nuntium/middleware.py

```diff
@@ -0,0 +1,33 @@
+from django.middleware.locale import LocaleMiddleware
+from django.utils import translation
+from django.conf import settings
+
+from nuntium.models import WriteItInstanceConfig
+
+
+class InstanceLocaleMiddleware(LocaleMiddleware):
+    def process_request(self, request):
+        check_path = self.is_language_prefix_patterns_used()
+        try:
+            default_language = WriteItInstanceConfig.objects.get(writeitinstance__slug=request.subdomain).default_language
+        except WriteItInstanceConfig.DoesNotExist:
+            default_language = settings.LANGUAGE_CODE
+
+        # Django 1.6 doesn't provide a nice way to hook into the language
+        # selection process to override it with a default. So instead we add
+        # a default language to the session and then remove it again after
+        # running 'get_language_from_request'.
+        did_override_session_language = False
```

Comment:
I'm assuming you want this "default" to take precedence to a user's browser setting, but still overrideable by the URL; in which case, this is okay, though I wonder if it might be clearer to copy the first bits of `get_language_from_request` into here? Then if there's a language in the URL it doesn't have to do any database look up. Something like:

```python
def process_request(self, request):
   """Same as parent, except calling our own get_language_from_request"""
    check_path = self.is_language_prefix_patterns_used()
    language = self.get_language_from_request(
        request, check_path=check_path)
    translation.activate(language)
    request.LANGUAGE_CODE = translation.get_language()

def get_language_from_request(request, check_path=False):
    from django.conf import settings
    supported = SortedDict(settings.LANGUAGES)

    if check_path:
        lang_code = translation.get_language_from_path(request.path_info, supported)
        if lang_code is not None:
            return lang_code

    try:
        lang_code = WriteItInstanceConfig.objects.get(writeitinstance__slug=request.subdomain).default_language
    except WriteItInstanceConfig.DoesNotExist:
        lang_code = None

    # Same behaviour as session entry – (Django aside: not sure why this doesn't use get_supported_language_variant like cookie does?)
    if lang_code in supported and lang_code is not None and translation.check_for_language(lang_code):
        return lang_code

    # Call with check_path False as we've already done that above!
    return translation.get_language_from_request(request, check_path=False)
```